backend/internal/nucleusdb/json-models/models.go

```diff
@@ -132,29 +132,158 @@ type JobMapping struct {
 	Schema      string
 	Table       string
 	Column      string
-	Transformer string
+	Transformer *Transformer
 	Exclude     bool
 }
 
 func (jm *JobMapping) ToDto() *mgmtv1alpha1.JobMapping {
+
 	return &mgmtv1alpha1.JobMapping{
 		Schema:      jm.Schema,
 		Table:       jm.Table,
 		Column:      jm.Column,
-		Transformer: jm.Transformer,
+		Transformer: jm.Transformer.ToDto(),
 		Exclude:     jm.Exclude,
 	}
 }
 
 func (jm *JobMapping) FromDto(dto *mgmtv1alpha1.JobMapping) error {
+
+	t := &Transformer{}
+	if err := t.FromDto(dto.Transformer); err != nil {
+		return err
+	}
 	jm.Schema = dto.Schema
 	jm.Table = dto.Table
 	jm.Column = dto.Column
-	jm.Transformer = dto.Transformer
+	jm.Transformer = t
 	jm.Exclude = dto.Exclude
 	return nil
 }
 
+type Transformer struct {
+	Value  string
+	Config *TransformerConfigs
+}
+
+type TransformerConfigs struct {
+	EmailConfig *EmailConfigs
+	FirstName   *FirstNameConfig
+	Uuidv4      *Uuidv4Config
+	PhoneNumber *PhoneNumberConfig
+	Passthrough *PassthroughConfig
```

Comment:
Since you already did this, it's fine - but we can most likely get away with only needing configs for transformers that actually require it. Having them be empty configs isn't really doing much.

Reply:
yeah i left them blank for right now but intending on filling them up later, likely these will all have some configs besides the passthrough one